### DISCUSSION TITLE
Nvidia's new (?) ota update process

### DISCUSSION THREAD

**Discussion by bhagen55** [OE4T/meta-tegra#843] [General]
Nvidia seems to have a new method for conducting OTA updates that appears to me to be almost fully standalone (doesn't require tools to already be built into the image)
https://docs.nvidia.com/jetson/l4t/index.html#page/Tegra%20Linux%20Driver%20Package%20Development%20Guide/updating_jetson_and_host.html

Has anyone else taken a peek at this? At first glance it seems to be a much more robust means of updating the entire system.

Compared to [the documented process](https://github.com/OE4T/meta-tegra/wiki/Over-the-air-reflashing-process), the steps to take on the device itself are much simpler (just extract and run a script).

I am puzzling over the instructions to create the OTA payload. They are of course oriented to people using the more standard L4T packages downloaded directly from Nvidia rather than generated by Yocto/OE.

**Comment by madisongh** [comment]
If there's stuff you find there that can be applied generally, and not just to running stock L4T/JetPack, especially if it would either simplify the process or make it more robust, then please share.  Likewise if you have any other improvements for the example process on the wiki page you point to.  I could see using the `kernel-bootctrl` settings to boot into the recovery kernel/dtb/rootfs being helpful, at least for some use cases, for instance.

**Reply by bhagen55** [reply]
I'm attempting to hack my way to getting the process to work for an OE4T image right now in an informal sense. Once I get my head wrapped around how it works I'll see about formalizing how to apply it/use it (if it works :) )
